### PR TITLE
fixed saving/loading core settings #2714

### PR DESCRIPTION
**Pull request by MichaelPetrinolis**
change callbackpath setting to PathString
added OrchardCore.Registration to dependencies
display an error when facebook login succeeds but site does not allow registration

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Facebook/Settings/FacebookLoginSettings.cs [hunk 1]
```diff
@@ -1,7 +1,9 @@
+using Microsoft.AspNetCore.Http;
+
 namespace OrchardCore.Facebook.Settings
 {
     public class FacebookLoginSettings
     {
-        public string CallbackPath { get; set; }
+        public PathString CallbackPath { get; set; }
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Facebook/Settings/FacebookLoginSettings.cs]: Shouldn't we use `/signin-facebook` as the default value here as this is be what users want most of the time?

**Comment by sebastienros** [comment]
Did some minor updates you'll need to review again ;) You might even be thinking that I am making your contributions worse!

File: src/OrchardCore.Modules/OrchardCore.Facebook/Drivers/FacebookCoreSettingsDisplayDriver.cs [hunk 2]
```diff
@@ -57,47 +57,47 @@ public override async Task<IDisplayResult> EditAsync(FacebookCoreSettings settin
...
+                {
+                    model.AppSecret = protector.Unprotect(settings.AppSecret);
+                }
 
             }).Location("Content:0").OnGroup(FacebookConstants.Features.Core);
         }
 
         public override async Task<IDisplayResult> UpdateAsync(FacebookCoreSettings settings, BuildEditorContext context)
         {
-            if (context.GroupId == FacebookConstants.Features.Core)
```
Comment (MichaelPetrinolis) [inline src/OrchardCore.Modules/OrchardCore.Facebook/Drivers/FacebookCoreSettingsDisplayDriver.cs]: If we remove this check, the UpdateAsync executes when saving the FaceBookLoginSettings and sets validation errors in the login settings view

**Comment by PinpointTownes** [comment]
> added OrchardCore.Registration to dependencies

That would be great if we could avoid depending on the Users module. Ideally, the social providers modules (like FB) should be usable as a standalone module since the ultimate plan is to support external authentication-only scenarios, where you don't have a local users DB (and thus, no Users-like module).

**Comment by MichaelPetrinolis** [comment]
The only reason I added is for the end user to easily allow New Users Registration. Will remove it and add to documentation

**Comment by MichaelPetrinolis** [comment]
Sorry for the mess, minor refactorings and a full UAT test. It seems to work as expected. Thanx for the reviews!!!

**Pull request merged**